.github/actions/bazel/action.yaml

```diff
@@ -70,6 +73,55 @@ runs:
           ${{ steps.metrics-tmpdir.outputs.dir }}/bazel-bep-*.pb.gpg
           ${{ steps.metrics-tmpdir.outputs.dir }}/profile-*.json
 
+
+      # Remove unnecessary entries from the JSON execlog and create a CSV out of it
+    - name: Clean up execlog
+      shell: bash
+      if: (success() || failure()) && inputs.execlog-artifact-name != ''
+      run: |
+        execlog_json_in='${{ steps.metrics-tmpdir.outputs.dir }}/execlogs.json'
+        find '${{ steps.metrics-tmpdir.outputs.dir }}' -name 'execlog-*.json' \
+          -exec cat {} \; > "$execlog_json_in"
+
+        execlog_csv_out='${{ steps.metrics-tmpdir.outputs.dir }}/execlogs.csv'
+        if ! [ -e "$execlog_json_in" ]; then
+          echo "no execlog found"
+          exit 0
+        fi
+
+        # this filters out some entries from the input JSON execlog. Without this,
+        # jq takes 3+ minutes to parse the execlog.
+        # Effectively removes a full line of "commandArgs", and takes care of skipping
+        # multiple lines for some multiline array values we don't care about
+        prog='
+          BEGIN { arr = 0; }
+          arr == 1 && /  }],?/ { arr = 0; next }
+          arr == 1 { next }
+          /  "commandArgs":.*/ { next }
+          /  "environmentVariables": \[\{/ { arr = 1; next; }
+          /  "inputs": \[\{/ { arr = 1; next; }
+          { print }
+          '
```

Comment:
discussed offline with @basvandijk : this is very brittle but we'll treat this as a blackbox for now and ASAP will replace it with an alternative that uses & parses `--execution_log_compact_file`